apps/src/music/blockly/MusicBlocklyWorkspace.ts

```diff
@@ -444,6 +455,46 @@ export default class MusicBlocklyWorkspace {
     const codeCopy = JSON.parse(JSON.stringify(code));
 
     Blockly.serialization.workspaces.load(codeCopy, this.workspace);
+
+    if (this.toolbox?.addFunctionCalls && this.toolbox?.type === 'flyout') {
+      this.generateFunctionCallBlocks();
+    }
+  }
+
+  // For each function body in the current workspace, add a function call
+  // block to the toolbox.
+  generateFunctionCallBlocks() {
+    const allFunctions: GoogleBlockly.serialization.procedures.State[] = [];
+
+    (this.workspace?.getTopBlocks() as ProcedureBlock[])
+      .filter(block => block.type === 'procedures_defnoreturn')
+      .forEach(block => {
+        allFunctions.push(
+          Blockly.serialization.procedures.saveProcedure(
+            block.getProcedureModel()
+          )
+        );
+      });
```

Comment:
There are a couple other ways we could get this information without having to iterate over all top blocks. One way is to get the procedures directly from the map on the workspace.

```
this.workspace.getProcedureMap().getProcedures().map(
  proc=>Blockly.serialization.procedures.saveProcedure(proc)
)
```

Another way would be just serialize and grab the procedures from there:

```
Blockly.serialization.workspaces.save(this.workspace).procedures
```

Your way works too, so I'm just offering in case they're of interest.